File: Railflow.Core/Entities/Train.cs
namespace Railflow.Core.Entities;

public class Train
{
    public Guid Id { get; private set; }
    public int Number { get; private set; }
    public float? MaxSpeed { get; private set; }
    public int Capacity { get; private set; }

    public Train(Guid id, int number, float? maxSpeed, int capacity)
    {
        Id = id;
        Number = number;
        MaxSpeed = maxSpeed;
        Capacity = capacity;
    }
}

File: client-app/src/app/models/train.ts
export interface Train {
  id: string;
  number: number;
  maxSpeed: number | null;
  capacity: number;
}

export interface TrainFormValues {
  number: number;
  maxSpeed: number | null;
  capacity: number;
}

export class Train implements Train {
  constructor(init?: TrainFormValues) {
    Object.assign(this, init);
  }
}